src/main/java/com/carleton/CapstoneSystem/User.java

```diff
@@ -0,0 +1,86 @@
+package com.carleton.CapstoneSystem;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.*;
+/**
+ * @author Ibrahim ALi Fawaz
+ * this abstract class represents a general user of the fourth year project website. more specific roles of the user should be shown in the subclasses of the class
+ */
+@Entity
+@Inheritance
+public abstract class User {

```

Comment:
if you make this abstract, we cannot instantiate a user.

Reply:
we wont need to, we would only need to instantiate the other three types